pandas/tests/io/json/test_pandas.py

```diff
@@ -1601,3 +1601,10 @@ def test_json_indent_all_orients(self, orient, expected):
     def test_json_negative_indent_raises(self):
         with pytest.raises(ValueError, match="must be a nonnegative integer"):
             pd.DataFrame().to_json(indent=-1)
+
+    def test_frame_int_overflow(self):
+        # GH 30320
+        encoded_json = json.dumps([{"col": "31900441201190696999"}, {"col": "Text"}])
+        expected = DataFrame({"col": ["31900441201190696999", "Text"]})
```

Comment:
IIUC from the issue, the expected output is an object column containing a number and a string?

Reply:
They have written the number in a float form. I'll look into generating an object.

> 3.190044e+19

Reply:
I'm happy with the string not being coerced, although I think the issue _may_ need further discussion. see https://github.com/pandas-dev/pandas/issues/30320#issuecomment-567491918.

What happens now when the `encoded_json` is an `int`? Maybe we should test for this too here?

Reply:
@simonjayhawkins Please see provide guidance on this [comment](https://github.com/pandas-dev/pandas/issues/30320#issuecomment-567602918)

Reply:
> @simonjayhawkins Please see and provide guidance on this [comment](https://github.com/pandas-dev/pandas/issues/30320#issuecomment-567602918)

not sure about this. @WillAyd?

Reply:
I don't know the answer either. Probably needs to be done in C but if you see a way to make it happen in the error handler works as well